probe-rs-target/src/chip.rs

```diff
@@ -32,6 +32,25 @@ pub struct Chip {
     pub flash_algorithms: Vec<String>,
 }
 
+impl Chip {
+    /// Create a generic chip with the given name, a single core,
+    /// and no flash algorithm or memory map. Used to create
+    /// generic targets.
+    pub fn generic_arm(name: &str, core_type: CoreType) -> Self {
+        Chip {
+            name: name.to_string(),
+            part: None,
+            cores: vec![Core {
+                name: "core".to_string(),
```

Comment:
I think we used "main" for other single cores. Would you mind changing this? :)

Reply:
It was "core" before as well for the generic chips, but I can change it.

Reply:
Oh, I see. I don't mind.